### ISSUE TITLE
VSCode command to search through open files and open the chosen file in the current (or last-used) editor group

### ISSUE THREAD

**Issue by macintacos** [microsoft/vscode#128874]
TLDR; I would like to have the ability to invoke a command to open a searchable list of all of the open files in a given window of VSCode, de-duplicated, regardless of what "group" they are located in. When "selecting" the file, the file would then open in the current (or last-used) editor group.

----

Essentially, the problem I'm trying to solve mostly stems from the fact that I do not care for editor tabs at all. I turn off tabs in VSCode, and typically navigate around by just searching for the file I want and opening it in the current editor.

The problem you run into with this approach is that, when trying to search through the list of files that are currently open, VSCode seems to deeply care about _where_ that file is located (i.e. which group it's located in). The first problem that this introduces for me is that it shows the user duplicates of the file that is open in all groups.

All told, this is a minor issue and I would be fine with it, but what ends up happening is that when I have a bunch of split editors open and I put my cursor in the editor window I want the file to open in, if I use the command to search through just the open files in my editor, and that editor happens to be in a different group, VSCode navigates to the file that is already open in the other group and switches my input focus to that group. I find this to be behavior that I really don't want - I always want to open the file that I search for in the editor I'm currently in.

Perhaps if there was an option to always open files in the last editor that had focus, that'd solve my problem, however I would find it really valuable if there were a quick-pick list of all of the files I have open (listed only once, regardless of what group they're in), plus the ability to open that file in the editor I'm currently in. I don't like going to the file, I want the file to come to me (if that makes sense).

Please let me know if you need any further clarification on this. I find that this concept is hard to describe to folks who use tabs all the time when editing.

**Comment by macintacos** [comment]
I couldn't find an issue that described what I'm looking for, and none of those issue appear to be the same either.

**Comment by ArturoDent** [comment]
Just curious, is this setting disabled: `Workbench >Editor: Reveal if Open`.  The default is disabled.  When I <kbd>Ctrl</kbd>+<kbd>P</kbd> and select a file already opened in another editor group.  It opens another copy in the current group, it does not reveal the same file in some other group.

**Comment by macintacos** [comment]
@ArturoDent thanks for the pointer! `ctrl+p` is just the generic file picker though - I want to see a list of only the files that are currently open, not search through all of the files in my project (listed by order of recency).

Essentially, I want a de-duplicated list of open files from the "View: Show All Editors By Most Recently Used" view, that allows me to open any of the files in that list in the editor that currently has focus. This currently] doesn't seem to be possible.

**Comment by bpasero** [comment]
![image](https://user-images.githubusercontent.com/900690/127818830-3930ce58-fbb5-44ae-9cee-3060a373bccc.png)

**Comment by macintacos** [comment]
I'm a little confused about why this was closed. @bpasero The screenshot you provided proves my point doesn't it? It shows the same file, duplicated, in the picker, and I want an option to not show the file duplicated there. I just want to see a list of all the current files I have open, regardless of what group they're in, and be able to open the file I choose in the group I'm currently in. Asking this question in Stack Overflow won't really get me anywhere because as far as I know, this doesn't work in VSCode right now. 

Happy to be proven wrong about that though!

**Comment by bpasero** [comment]
Ah! Sorry, I missed the "deduplication" part.

**Comment by TylerLeonhardt** [comment]
Isn’t the deduplication covered by 

![0E695993-489C-40BB-9D4B-7BECE07F2F73](https://user-images.githubusercontent.com/2644648/135923501-5bc9f0f0-34e6-4637-91a8-c67935f7c226.jpeg)

Or am I missing something?

**Comment by macintacos** [comment]
@TylerLeonhardt I think that only shows you all of the "active" tabs in the group you're currently in? I want to see all of the tabs in all editor groups, and then open the file that I select in the current group that I'm in.

**Comment by TylerLeonhardt** [comment]
@macintacos gotcha. And `cmd/ctrl+P` doesn't work in this scenario because you only wanna look through open editors and not files in your workspace?

I ask this because `cmd/ctrl+P` seems to have the "open this in the current group" behavior that you're looking for... but the "this" is not quite right.

**Comment by macintacos** [comment]
Yep, exactly! There's no picker that I know of that'll allow me to look through just a list of the files I currently have open (regardless of group) and open one of those files in the current group. Having that available (even if it were one of those `edt xyz` combos) would be great.

**Comment by macintacos** [comment]
For onlookers - I was tired of waiting for a built-in method of doing this, so I made an extension for it instead: https://marketplace.visualstudio.com/items?itemName=excessive-dev.buffers

Still believe that this should be native functionality, so I will leave this open.
